.github/actions/docker-build/action.yml

```diff
@@ -0,0 +1,51 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0.
+
+name: smithy-rs Docker Build
+description: Run Docker build command for smithy-rs
+inputs:
+  action:
+    description: What action to run in the Docker build
+    required: true
+runs:
+  using: composite
+  steps:
+  - uses: actions/cache@v2
+    name: Gradle Cache
+    with:
+      path: |
+        gradle/caches
+        gradle/wrapper
+      key: ${{ runner.os }}-gradle-${{ hashFiles('gradle/caches/**/*', 'gradle/wrapper/**/*') }}
+      restore-keys: |
+        ${{ runner.os }}-gradle-
+      # Pinned to the commit hash of v1.3.0
+  - uses: Swatinem/rust-cache@842ef286fff290e445b90b4002cc9807c3669641
+    with:
+      sharedKey: ${{ runner.os }}-${{ github.job }}
+      target-dir: ./smithy-rs-target
+  - name: Get artifacts
+    uses: ./smithy-rs/.github/actions/get-artifacts
+  - name: Prepare build image
+    shell: bash
+    run: |
+      set -x
+      if [[ -d smithy-rs-base-image ]]; then
+        IMAGE_TAG="$(cd smithy-rs; git ls-files -s --full-name "tools" | git hash-object --stdin)"
+        docker load -i smithy-rs-base-image/smithy-rs-base-image
+        docker tag "smithy-rs-base-image:${IMAGE_TAG}" "smithy-rs-base-image:local"
+      fi
+      ALLOW_LOCAL_BUILD=false ./smithy-rs/tools/ci-build/acquire-build-image
```

Comment:
do you mind adding some comments on this? I'm staring at it, but not really groking the logic. I think it's:

"if `smithy-rs-base-image` exists" (although I don't understand when it would or would not exist": load and tag `:IMAGE_TAG` with `:local`

run "acquire-build-image" but don't allow it to actually run docker build?